File: sdk/src/main/java/com/meniga/sdk/models/feed/operators/MenigaFeedOperations.java
package com.meniga.sdk.models.feed.operators;

import com.meniga.sdk.helpers.Result;
import com.meniga.sdk.models.feed.MenigaFeed;
import com.meniga.sdk.models.feed.MenigaFeedItem;
import com.meniga.sdk.models.feed.MenigaScheduledEvent;

import org.joda.time.DateTime;

/**
 * Copyright 2017 Meniga Iceland Inc.
 */
public interface MenigaFeedOperations {

	Result<MenigaFeed> getFeed(DateTime from, DateTime to);

	Result<MenigaFeed> getFeed(DateTime from, DateTime to, int skip, int take);

	Result<MenigaFeedItem> getEvent(long id);

    Result<MenigaScheduledEvent> getScheduledEvent(long id);
}

File: sdk/src/main/java/com/meniga/sdk/models/feed/operators/MenigaFeedOperations.kt
/*
 * Copyright 2017-2021 Meniga Iceland Inc.
 */
package com.meniga.sdk.models.feed.operators

import com.meniga.sdk.helpers.Result
import com.meniga.sdk.models.feed.FeedFilter
import com.meniga.sdk.models.feed.MenigaFeed
import com.meniga.sdk.models.feed.MenigaScheduledEvent

interface MenigaFeedOperations {
    fun getFeed(feedFilter: FeedFilter): Result<MenigaFeed>
    fun getScheduledEvent(id: Long): Result<MenigaScheduledEvent>
}
